File: 7_Fibonacci/7.py
def generarElementos(n):
    # Agregar los 2 primeros elementos
    a = []
    a.append(0)
    a.append(1)
    for i in range(2, n):
        a.append(a[i - 1] + a[i - 2])
    return a

def mostrarSucesion(a, n):
    for i in range(0, n):
        print(str(a[i]), end=" ")
    print("\n")

n = int(input("# de elementos a generar (n >= 1): "))
a = generarElementos(n)
mostrarSucesion(a, n)

File: 7_Fibonacci/7.rb
def generarElementos(n)
    a = []
    # Se insertan los 2 primeros elementos
    a << 0
    a << 1
    2.upto(n-1) do |i|
        a << a[i - 1] + a[i - 2]
    end
    return a
end

def mostrarSucesion(a, n)
    0.upto(n-1) do |i|
        print "#{a[i]} "
    end
end

print "Número de elementos a generar: "
n = gets.chomp.to_i
a = generarElementos(n)
mostrarSucesion(a, n)
puts
